tests/pyright_test.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+import os
+import subprocess
+import sys
+from pathlib import Path
+from typing import Iterable, List
+
+
+_WELL_KNOWN_FILE = Path("tests", "pyright_test.py")
+_PYRIGHT_COMMAND = ["npx", "-p", "pyright@1.1.113", "pyright"]
```

Comment:
Should this be unpinned, if mypy is getting run at master (based on the requirements file, anyway)?

Reply:
Ah, I see #5048 says this is probably just a starter :)

Reply:
I think we'd prefer a pinned version, so our CI doesn't start failing with no action. mypy is currently run from master because of transitional state around the recent directory reorganization.

Reply:
Pinning mypy was discussed in #4332. I am still in favor of it (and we should also pin pytest while we are at it). As Jelle pointed out, using master has the disadvantage that build can fail for reasons unrelated to a PR.